sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -393,36 +420,268 @@ def tune(
                     )
 
         # Add metrics collector to the Katib Experiment.
-        # Up to now, we only support parameter `kind`, of which default value
-        # is `StdOut`, to specify the kind of metrics collector.
+        # Up to now, We only support parameter `kind`, of which default value is
+        # `StdOut`, to specify the kind of metrics collector.
         experiment.spec.metrics_collector_spec = models.V1beta1MetricsCollectorSpec(
-            collector=models.V1beta1CollectorSpec(kind=metrics_collector_config["kind"])
+            collector=models.V1beta1CollectorSpec(
+                kind=metrics_collector_config["kind"]
+            ),
+            source=models.V1beta1SourceSpec(
+                filter=models.V1beta1FilterSpec(
+                    metrics_format=[
+                        # For example: train_loss=0.846
+                        r"([\w|-]+)\s*=\s*([+-]?\d*(\.\d+)?([Ee][+-]?\d+)?)",
+                        # For example: 'train_loss':0.846
+                        r"'([\w|-]+)'\s*:\s*([+-]?\d*(\.\d+)?([Ee][+-]?\d+)?)",
+                    ]
+                )
+            ),
```

Comment:
Should we modify the metrics collector format only when user sets trainer_parameters, dataset_parameters, and model_parameters ? We don't want to modify it when user sets `objective`.

Reply:
Got it! I will change according to your suggestions.